src/backend/gl/src/device.rs

```diff
@@ -1568,26 +1550,100 @@ impl d::Device<B> for Device {
         fence: &n::Fence,
         timeout_ns: u64,
     ) -> Result<bool, d::OomOrDeviceLost> {
-        if !self.share.private_caps.sync {
-            return Ok(true);
```

Comment:
So one thing this patch does right now is *remove* this line.  I am not sure why this is there, but the behavior beforehand was *not* actually to call `glFlush` on waiting for a fence with no sync capability due to this early return.  If you want, I can keep this behavior and just eliminate `FenceInner::Pending(None)` as a state entirely (without sync capability, fences could just be created `FenceInner::Idle { signaled: true }`.

Reply:
I think this is too late for this check anyway. If "sync" capability is not available, we'd not be able to create a `Pending` sync in the first place. So removing the check from here is fine.

Reply:
With this patch, if "sync" is not available the gl backend will create a `FenceInner::Pending(None)` in submit, so it is possible to create a `FenceInner::Pending(None)` and hit the code below this (which is not from this patch).  Beforehand, I *think* it was not possible to hit the code below.